Marlin/src/feature/mmu3/ultralcd.cpp

```diff
@@ -0,0 +1,225 @@
+    lcd_put_lchar(0, 3, selected == LCD_LEFT_BUTTON_CHOICE ? '>' : ' ');
+    lcd_put_u8str(first_choice);
+    lcd_put_lchar(second_col, 3, selected == LCD_MIDDLE_BUTTON_CHOICE ? '>' : ' ');
+    lcd_put_u8str(second_choice);
+    if (third_choice) {
+      lcd_put_lchar(18, 3, selected == LCD_RIGHT_BUTTON_CHOICE ? '>' : ' ');
+      lcd_put_u8str(third_choice);
+    }
+  }
+
+  void lcd_space(uint8_t n) {
+    while (n--) lcd_put_lchar(' ');
+  }
+
+  // Print extruder status (5 chars total)
+  // Scenario 1: "F?"
+  //              There is no filament loaded and no tool change is in progress
+  // Scenario 2: "F[nr.]"
+  //              [nr.] ranges from 1 to 5.
+  //              Shows which filament is loaded. No tool change is in progress
+  // Scenario 3: "?>[nr.]"
+  //              [nr.] ranges from 1 to 5.
+  //              There is no filament currently loaded, but [nr.] is currently being loaded via tool change
+  // Scenario 4: "[nr.]>?"
+  //              [nr.] ranges from 1 to 5.
+  //              This scenario indicates a bug in the firmware if ? is on the right side
+  // Scenario 5: "[nr1.]>[nr2.]"
+  //              [nr1.] ranges from 1 to 5.
+  //              [nr2.] ranges from 1 to 5.
+  //              Filament [nr1.] was loaded, but [nr2.] is currently being loaded via tool change
+  // Scenario 6: "?>?"
+  //              This scenario should not be possible and indicates a bug in the firmware
+  uint8_t lcdui_print_extruder(void) {
+    uint8_t chars = 1;
+    lcd_space(1);
+    if (MMU2::mmu2.get_current_tool() == MMU2::mmu2.get_tool_change_tool()) {
+      lcd_put_lchar('F');
+      lcd_put_lchar(MMU2::mmu2.get_current_tool() == (uint8_t)MMU2::FILAMENT_UNKNOWN ? '?' : MMU2::mmu2.get_current_tool() + '1');
+      chars += 2;
+    }
+    else {
+      lcd_put_lchar(MMU2::mmu2.get_current_tool() == (uint8_t)MMU2::FILAMENT_UNKNOWN ? '?' : MMU2::mmu2.get_current_tool() + '1');
+      lcd_put_lchar('>');
+      lcd_put_lchar(MMU2::mmu2.get_tool_change_tool() == (uint8_t)MMU2::FILAMENT_UNKNOWN ? '?' : MMU2::mmu2.get_tool_change_tool() + '1');
+      chars += 3;
+    }
+    return chars;
+  }
+
+  bool pgm_is_whitespace(const char *c_addr) {
+    const char c = pgm_read_byte(c_addr);
+    return c == ' ' || c == '\t' || c == '\r' || c == '\n';
+  }
+
+  bool pgm_is_interpunction(const char *c_addr) {
+    const char c = pgm_read_byte(c_addr);
+    return c == '.' || c == ',' || c == ':' || c == ';' || c == '?' || c == '!' || c == '/';
+  }
+
+
+  /**
+   * @brief show full screen message
+   *
+   * This function is non-blocking
+   * @param msg message to be displayed from PROGMEM
+   * @return rest of the text (to be displayed on next page)
+   */
+  static const char* lcd_display_message_fullscreen_nonBlocking_P(const char *msg) {
+    const char *msgend = msg;
+    bool multi_screen = false;
```

Comment:
This variable is assigned but never used. Remove it and get rid of commented out code thats using it.

```suggestion
```